games-roguelike/angband/patches/angband-4.1.2.patchset

```diff
@@ -0,0 +1,36 @@
+From e2a4f92d3e66bde0686073e709a0e940ffb5be6b Mon Sep 17 00:00:00 2001
+From: begasus <begasus@gmail.com>
+Date: Mon, 21 May 2018 23:14:49 +0200
+Subject: import partial patch from previous version
+
+
+diff --git a/configure.ac b/configure.ac
+index 94ef3fe..66fdf2c 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -98,7 +98,7 @@ CFLAGS="$CFLAGS -DHAVE_CONFIG_H"
+ CPPFLAGS="$CPPFLAGS -I." 
+ 
+ if test "$GCC" = "yes"; then
+-	CFLAGS="$CFLAGS -W -Wall -Wextra -Wno-unused-parameter -pedantic"
++	CFLAGS="$CFLAGS -W -Wall"
+ 	AC_MSG_CHECKING([if gcc supports -Wno-missing-field-initializers])
+ 	_gcc_cflags_save=$CFLAGS
+ 	CFLAGS="-Wno-missing-field-initializers"
+diff --git a/src/z-rand.c b/src/z-rand.c
+index 57f0e4b..526af98 100644
+--- a/src/z-rand.c
++++ b/src/z-rand.c
+@@ -537,7 +537,9 @@ void rand_fix(u32b val)
+ 	rand_fixval = val;
+ }
+ 
++#ifndef __HAIKU__
+ int getpid(void);
```

Comment:
Actually, just changing `int` to `pid_t` should solve the problem for us and possibly others, as that's the [correct signature][0] for the function :)

```diff
-+#ifndef __HAIKU__
- int getpid(void);
+-int getpid(void);
++pid_t getpid(void);
-+#endif
```
[0]: http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpid.html